source/solvers/cahn_hilliard.cc

```diff
@@ -730,8 +730,8 @@ CahnHilliard<dim>::postprocess(bool first_iteration)
 
       if (multiphysics->fluid_dynamics_is_block())
         {
-          if (this->simulation_parameters.multiphysics
-                .use_time_average_velocity_field &&
+          if (this->simulation_parameters.initial_condition->type ==
+                Parameters::InitialConditionType::average_velocity_profile &&
```

Comment:
I am unsure I understand what this check is for, but I feel also maybe it's because I am not seeing the big picture. COuld be worth it to add a comment here (and in the other auxiliary physics also)

Reply:
Also, if we reset time to 0, is this condition true when it should be?